test/e2e/common/node/pod_level_resources.go

```diff
@@ -111,59 +100,23 @@ func makeObjectMetadata(name, namespace string) metav1.ObjectMeta {
 
 type containerInfo struct {
 	Name      string
-	Resources *resourceInfo
-}
-type resourceInfo struct {
-	CPUReq string
-	CPULim string
-	MemReq string
-	MemLim string
+	Resources *e2epod.ContainerResources
 }
 
 func makeContainer(info containerInfo) v1.Container {
-	cmd := []string{"/bin/sh", "-c", "sleep 1d"}
-	res := getResourceRequirements(info.Resources)
+	var resources v1.ResourceRequirements
+	if info.Resources != nil {
+		resources = *info.Resources.ResourceRequirements()
+	}
 	return v1.Container{
 		Name:      info.Name,
 		Command:   cmd,
-		Resources: res,
+		Resources: resources,
 		Image:     imageutils.GetE2EImage(imageutils.BusyBox),
-		VolumeMounts: []v1.VolumeMount{
-			{
-				Name:      "sysfscgroup",
-				MountPath: mountPath,
-			},
-		},
-	}
-}
-
-func getResourceRequirements(info *resourceInfo) v1.ResourceRequirements {
-	var res v1.ResourceRequirements
-	if info != nil {
-		if info.CPUReq != "" || info.MemReq != "" {
-			res.Requests = make(v1.ResourceList)
-		}
-		if info.CPUReq != "" {
-			res.Requests[v1.ResourceCPU] = resource.MustParse(info.CPUReq)
-		}
-		if info.MemReq != "" {
-			res.Requests[v1.ResourceMemory] = resource.MustParse(info.MemReq)
-		}
-
-		if info.CPULim != "" || info.MemLim != "" {
-			res.Limits = make(v1.ResourceList)
-		}
-		if info.CPULim != "" {
-			res.Limits[v1.ResourceCPU] = resource.MustParse(info.CPULim)
-		}
-		if info.MemLim != "" {
-			res.Limits[v1.ResourceMemory] = resource.MustParse(info.MemLim)
-		}
 	}
-	return res
 }
 
-func makePod(metadata *metav1.ObjectMeta, podResources *resourceInfo, containers []containerInfo) *v1.Pod {
+func makePod(metadata *metav1.ObjectMeta, podResources *e2epod.ContainerResources, containers []containerInfo) *v1.Pod {
```

Comment:
Can we dedup this? The logic is similar here: https://github.com/kubernetes/kubernetes/blob/40f222b6201d5c6476e5b20f57a4a7d8b2d71845/test/e2e/framework/pod/resize.go#L160-L198